### DISCUSSION TITLE
Host public proxy?

### DISCUSSION THREAD

**Discussion by acheong08** [acheong08/EdgeGPT#194] [Polls]
- #178

**Comment by Harry-Jing** [comment]
as option？

**Reply by acheong08** [reply]
Yes

**Reply by acheong08** [reply]
It will always be optional and you can host it yourself with https://github.com/acheong08/EdgeGPT-Proxy

**Reply by Nu11Const** [reply]
And wss link also can as a option.At the weekend I will make a Pull Request.

**Reply by acheong08** [reply]
wss works without proxy

**Reply by Nu11Const** [reply]
> wss works without proxy

But with IPV6 ,it doesn't work

**Reply by Nu11Const** [reply]
I think it is possible to add `api` parameters during Chatbot initialization, and if there are no parameters, can use `BING_PROXY_URL` environment variables, and finally the official API.

**Comment by Nu11Const** [comment]
Can don't use this project,Use reverse proxy.

**Reply by acheong08** [reply]
This is a reverse proxy sort of.

**Comment by Lingpeixie** [comment]
Installed the latest version of Go, the command installation shows the following content:

C:\Users\admin>go install github.com/acheong08/EdgeGPT-Proxy@latest
# github.com/fvbock/endless
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:64:11: undefined: syscall.SIGUSR1
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:65:11: undefined: syscall.SIGUSR2
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:68:11: undefined: syscall.SIGTSTP
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:111:13: undefined: syscall.SIGUSR1
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:112:13: undefined: syscall.SIGUSR2
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:115:13: undefined: syscall.SIGTSTP
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:119:13: undefined: syscall.SIGUSR1
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:120:13: undefined: syscall.SIGUSR2
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:123:13: undefined: syscall.SIGTSTP
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:224:11: undefined: syscall.Kill
go\pkg\mod\github.com\fvbock\endless@v0.0.0-20170109170031-447134032cb6\endless.go:224:11: too many errors

**Reply by acheong08** [reply]
Windows not supported

**Reply by Lingpeixie** [reply]
> Windows not supported

Thank you for the answer.

**Reply by acheong08** [reply]
To support windows, replace endless and use gin to run the router. It removes graceful restarts but works fine

**Reply by Allen-c-HE** [reply]
> To support windows, replace endless and use gin to run the router. It removes graceful restarts but works fine

uh, how? too little knowlege on this, waiting for a version directly for windows.

**Comment by Allen-c-HE** [comment]
AWESOME!
THANKS!
________________________________
发件人: Antonio Cheong ***@***.***>
发送时间: 2023年3月25日 10:56
收件人: acheong08/EdgeGPT ***@***.***>
抄送: Allen-c-HE ***@***.***>; Comment ***@***.***>
主题: Re: [acheong08/EdgeGPT] Host public proxy? (Discussion #194)


wss works without proxy

―

**Comment by Nu11Const** [comment]
If this `https://edge.churchless.tech/edgesvc/turing/conversation/create` is the public proxy?This address doesn't work.

**Reply by acheong08** [reply]
What is the error? It worked for me

**Reply by Nu11Const** [reply]
404 not found error,but now is work again.

**Reply by DreamMaoMao** [reply]
> 404未找到错误，但现在又开始工作了。

i have the same problem
![image](https://user-images.githubusercontent.com/30348075/227699632-ba1f65e9-98f8-4eb6-9bd2-059333bb43ba.png)

![image](https://user-images.githubusercontent.com/30348075/227699647-11ba7ab9-72ea-4c92-a8f0-b941f4d0b141.png)
 i don't  understand how to use it

**Reply by Nu11Const** [reply]
> > 404未找到错误，但现在又开始工作了。
> 
> i have the same problem ![image](https://user-images.githubusercontent.com/30348075/227699632-ba1f65e9-98f8-4eb6-9bd2-059333bb43ba.png)
> 
> ![image](https://user-images.githubusercontent.com/30348075/227699647-11ba7ab9-72ea-4c92-a8f0-b941f4d0b141.png) i don't understand how to use it

环境变量不是`https_proxy`，是`BING_PROXY_URL`

**Reply by DreamMaoMao** [reply]
> > > 404未找到错误，但现在又开始工作了。
> > 
> > 
> > i have the same problem ![image](https://user-images.githubusercontent.com/30348075/227699632-ba1f65e9-98f8-4eb6-9bd2-059333bb43ba.png)
> > ![image](https://user-images.githubusercontent.com/30348075/227699647-11ba7ab9-72ea-4c92-a8f0-b941f4d0b141.png) i don't understand how to use it
> 
> 环境变量不是`https_proxy`，是`BING_PROXY_URL`

![image](https://user-images.githubusercontent.com/30348075/227701557-974b8e6d-2e17-4dbb-9bbb-e50ee2ed35f0.png)
![image](https://user-images.githubusercontent.com/30348075/227701614-157778e9-d5be-41d8-84e5-67ef7875f3d2.png)
![image](https://user-images.githubusercontent.com/30348075/227701630-41e78732-a710-456c-a35e-de10afd36020.png)

设置了这个环境变量了  还是没有用

**Reply by DreamMaoMao** [reply]
> > > 404未找到错误，但现在又开始工作了。
> > 
> > 
> > i have the same problem ![image](https://user-images.githubusercontent.com/30348075/227699632-ba1f65e9-98f8-4eb6-9bd2-059333bb43ba.png)
> > ![image](https://user-images.githubusercontent.com/30348075/227699647-11ba7ab9-72ea-4c92-a8f0-b941f4d0b141.png) i don't understand how to use it
> 
> 环境变量不是`https_proxy`，是`BING_PROXY_URL`

我的问题 我一直是直接调用pypi模块的 现在直接下载源码 后设置环境变量export BING_PROXY_URL=“https://edge.churchless.tech/edgesvc/turing/conversation/create” 就可以用了

**Reply by Nu11Const** [reply]
其实不用export，最新版源码已经默认了这个选项。
不过这个地址是cf cdn的，国内访问还是挺慢的。

**Reply by DreamMaoMao** [reply]
> 其实不用export，最新版源码已经默认了这个选项。 不过这个地址是cf cdn的，国内访问还是挺慢的。

我在代码上看到了 不用export其实也是可以的  但是代码改动我这里有个建议，就是改动过的时候希望不要频繁改动模块的接口参数，
尽可能在模块内部变动，这样通过调用EdgeGPT模块来实现业务逻辑的不用频繁改动代码来实现接口兼容，只需要进行模块拉取更新

**Reply by Nu11Const** [reply]
> > 其实不用export，最新版源码已经默认了这个选项。 不过这个地址是cf cdn的，国内访问还是挺慢的。
> 
> 我在代码上看到了 不用export其实也是可以的 但是代码改动我这里有个建议，就是改动过的时候希望不要频繁改动模块的接口参数， 尽可能在模块内部变动，这样通过调用EdgeGPT模块来实现业务逻辑的不用频繁改动代码来实现接口兼容，只需要进行模块拉取更新

这个没有办法，微软一直在想尽办法禁止中国大陆访问新必应，接口一直在变动
